### PR TITLE
WIP: Debian stretch support

### PR DESCRIPTION
**Pull request by helmo**
I'm still working on a fix for 'Access denied for user 'root'@'localhost'

### CODE REVIEW

**Comment by helmo** [comment]
This includes the work from #309

**Comment by helmo** [comment]
I wanted to add something like this, based on the Aegir Debian packages.... problem is mysql is not installed that early.
Adding 'apt install default-mysql-server --yes' in the same if might be a solution ... trying now.

```patch
diff --git a/install.sh b/install.sh
index 9682d82..21eca4f 100755
--- a/install.sh
+++ b/install.sh
@@ -284,16 +284,35 @@ if [ "$TRAVIS" == "true" ]; then
   MAKEFILE_PATH="https://raw.githubusercontent.com/opendevshop/devshop/$DEVSHOP_VERSION/build-devmaster.make"
 fi
 
-if [ -f '/root/.my.cnf' ]
+# Determin if we create a db accoount or ask for it.
+# Starting from MariaDB 10.1... https://www.drupal.org/node/2770819
+# TODO: Work with future versions
+if mysql --host=$AEGIR_DB_HOST --database=mysql --execute="show tables;" > /dev/null 2>&1
 then
-  MYSQL_ROOT_PASSWORD=$(awk -F "=" '/pass/ {print $2}' /root/.my.cnf)
-  echo " Password found at /root/.my.cnf, using $MYSQL_ROOT_PASSWORD"
+  # no password needed to login to the database ... MariaDB => 10.1 on Debian has this as default.
+  AEGIR_DB_USER="aegir_root"
+  MYSQL_ROOT_USER=$AEGIR_DB_USER;
+
+  # Random password, will be stored in /var/aegir/.drush/server_localhost.alias.drushrc.php
+  AEGIR_DB_PASS=$(openssl rand -base64 12)
+  MYSQL_ROOT_PASSWORD=$AEGIR_DB_PASS;
+
+  /usr/bin/mysql -e "GRANT ALL ON *.* TO '$AEGIR_DB_USER'@'$AEGIR_DB_HOST' IDENTIFIED BY '$AEGIR_DB_PASS' WITH GRANT OPTION"
+
 else
-  MYSQL_ROOT_PASSWORD=$(< /dev/urandom tr -dc _A-Z-a-z-0-9 | head -c${2:-32};echo;)
-  echo " Generating new MySQL root password... $MYSQL_ROOT_PASSWORD"
-  echo $MYSQL_ROOT_PASSWORD > /tmp/mysql_root_password
+
+       if [ -f '/root/.my.cnf' ]
+       then
+               MYSQL_ROOT_PASSWORD=$(awk -F "=" '/pass/ {print $2}' /root/.my.cnf)
+               echo " Password found at /root/.my.cnf, using $MYSQL_ROOT_PASSWORD"
+       else
+               MYSQL_ROOT_PASSWORD=$(< /dev/urandom tr -dc _A-Z-a-z-0-9 | head -c${2:-32};echo;)
+               echo " Generating new MySQL root password... $MYSQL_ROOT_PASSWORD"
+               echo $MYSQL_ROOT_PASSWORD > /tmp/mysql_root_password
+       fi
 fi
 
+
 echo $LINE
 echo " Hostname: $HOSTNAME_FQDN"
 echo " MySQL Root User: $MYSQL_ROOT_USER"
```

**Comment by helmo** [comment]
I added a CI test ... not sure what this error means . Any idea @jonpugh ?

```
Status: Downloaded newer image for geerlingguy/docker-debian9-ansible:latest
abb69b07537ac6b6bb275229c6261a3b71e01d64e6000c24b22e287c1e799f37
docker: Error response from daemon: oci runtime error: container_linux.go:265: starting container process caused "exec: \"/sbin/init\": stat /sbin/init: no such file or directory".
```

**Comment by jonpugh** [comment]
There's logic in the Robofile to change the docker run based on the image name.

**Comment by jonpugh** [comment]
> fatal: [devshop.local.computer]: FAILED! => {"changed": false, "msg": "Command failed rc=1, out=, err=Failed to create bus connection: No such file or directory\n"}

Debian based systems need dbus installed for ansible hostname module to work...

**Comment by jonpugh** [comment]
Which is weird, because I did already add it to https://github.com/opendevshop/ansible-role-devmaster ...

**Comment by helmo** [comment]
The log also shows it being installed two tasks up... could Debian use a different config? different socket file?

The travis detection if is also strange ,... the `if [ "$TRAVIS" == "true" ];` you use in multiple places executes as '[' 1 == true ']' so does it ever get executed? Or can it safely be removed.

**Comment by jonpugh** [comment]
Maybe it needs something different. Needs Googling ...

https://github.com/ansible/ansible/issues/25543

**Comment by jonpugh** [comment]
> does it ever get executed? Or can it safely be removed.

A PR would help determine that.

File: install.sh [hunk 1]
```diff
@@ -277,16 +287,59 @@ elif [ $OS == 'centos' ] || [ $OS == 'redhat' ] || [ $OS == 'fedora'  ]; then
...
-  echo " Generating new MySQL root password... $MYSQL_ROOT_PASSWORD"
+# Generate MySQL Password
+if [ "$TRAVIS" == "true" ]; then
+  echo "TRAVIS DETECTED! Setting empty 'root' user password."
+  MYSQL_ROOT_PASSWORD=''
   echo $MYSQL_ROOT_PASSWORD > /tmp/mysql_root_password
+  MAKEFILE_PATH="https://raw.githubusercontent.com/opendevshop/devshop/$DEVSHOP_VERSION/build-devmaster.make"
 fi
 
+# Determin if we create a db accoount or ask for it.
```
Comment (jonpugh) [inline install.sh]: Spelling

File: install.sh [hunk 2]
```diff
@@ -277,16 +287,59 @@ elif [ $OS == 'centos' ] || [ $OS == 'redhat' ] || [ $OS == 'fedora'  ]; then
...
+  set -x
+  if [ "$TRAVIS" == "1" ]; then
+    # By default this policy file prevents starting daemons, which is useful during docker builds but not in this case.
+    rm /usr/sbin/policy-rc.d
+  fi
+  # Install MariaDB early to be able to create an account for Aegir.
+  apt-get install default-mysql-server --yes
+
+  # no password needed to login to the database ... MariaDB => 10.1 on Debian has this as default.
+  AEGIR_DB_USER="aegir_root"
```
Comment (jonpugh) [inline install.sh]: Make sure AEGIR_DB_USER is being set by default, as it's used down below in the Ansible variables.

File: install.sh [hunk 3]
```diff
@@ -277,16 +287,59 @@ elif [ $OS == 'centos' ] || [ $OS == 'redhat' ] || [ $OS == 'fedora'  ]; then
...
+  AEGIR_DB_HOST="localhost"
+  MYSQL_ROOT_USER=$AEGIR_DB_USER;
+
+  # Random password, will be stored in /var/aegir/.drush/server_localhost.alias.drushrc.php
+  AEGIR_DB_PASS=$(openssl rand -base64 12)
```
Comment (jonpugh) [inline install.sh]: Look at line 331, just below. We are already generating the MYSQL_ROOT_PASSWORD variable with /dev/urandom...

Not sure which is best, but it should be consistent.

**Review by jonpugh — COMMENTED** [review]
Just a few changes needed. Once tests pass, this will be merged.

**Pull request closed without merging**